frontend/src/modules/Metrics/Components/Metrics.tsx

```diff
@@ -36,12 +36,18 @@ export const Metrics = () => {
       isDateInThisWeek(membership.submissionTime)
     )
   )
+  const uniqueStudentsInWeek = []
+  for (let i = 0; i < numStudentsInWeek.length; i++) {
+    if (numStudentsInWeek[i].length === students[i].groups.length) {
+      uniqueStudentsInWeek.push(students[i])
+    }
+  }
```

Comment:
I think that this section could be simplified with the use of the `filter()` function:
`const uniqueStudentsInWeek = numStudentsInWeek.filter((student, i) => student.length === students[i].groups.length);`

The only catch to this method is that you get an array of type `GroupMembership[] ` instead of `Student`. If you really do need the array to contain the actual students, using `forEach()` on the `numStudentsInWeek` array will allow you to do the same thing you did here but with less code:

`const uniqueStudentsInWeek = []`
`numStudentsInWeek.forEach((student, i) => {
    if (student.length === students[i].groups.length) {
      uniqueStudentsInWeek.push(students[i])
    }
  })`